### ISSUE TITLE
ValueError: dict contains fields not in fieldnames: 'unique_data_id' when rebuilding local database with new data.

### ISSUE THREAD

**Issue by jkwening** [codefordc/housing-insights#243]
I may be mistaken and hopefully it is only on my end but there seems to be an issue with the loading data work flow. I was getting a "ValueError: dict contains fields not in fieldnames: 'unique_data_id'" when I run load_data.py after following the steps in adding data sources guide.

After tracing through the code, the error is being thrown from the write() method in the CSVWriter module. It is expecting a 'unique_data_id' field but there wasn't one in the json data for my new data. So my guess is that as part of the workflow this should be automatically included when draft json is generated. 

That seems to resolve the problem on my end - I added code in make_draft_json.py to do so each time. I've committed the code change on the following branch on my cloned repo: 237-BROWSE-Ingestion-Infra for reference.

@speedyturkey @NealHumphrey I think it is related to the updates merged from issue  #176, so hoping either one of you might have insight into the error I ran into and whether my solution is the right one. Thanks.

**Comment by NealHumphrey** [comment]
@jkwening yep that's from #176. I think that perhaps we forgot to update the make_draft_json code  to also add that new unique_data_id field. In `/python/scripts/add_required_in_source_to_meta.py` that did this for the existing code so I had not yet tested it with a new data source. 

Is that the only fix in that branch? If you can make a pull request we can get that hotfix into the live code so other people don't have issues with it, and I can make sure it looks like a complete fix.

**Comment by jkwening** [comment]
This is not the only fix in the branch. As I traced through the code, I added doc-strings to the modules and their methods and minor refactoring to facilitate understanding the code base. The main goal of this was to ease debugging for others if they run into issue and expedite understanding of the code base and work flow. So I think they should be included in the pull request but someone should make sure my documentation is correct. In the meantime, I'll create an pull request for just the update to make_draft_json.py. Thanks.
